### PR TITLE
Wrapper service for any command

### PR DESCRIPTION
**Pull request by nagyrobi**
Solution for https://github.com/HASwitchPlate/openHASP-custom-component/issues/50.

### CODE REVIEW

File: custom_components/openhasp/__init__.py [hunk 1]
```diff
@@ -504,6 +518,14 @@ async def async_change_page(self, page):
         )
         self.async_write_ha_state()
 
+    async def async_command_wrapper(self, command, keyword, parameters):
+        """Sends commands directly to the plate entity (as a wrapper for MQTT commands sent to hasp/<nodename>/command)"""
+        cmd_topic = f"{self._topic}/" + command
+
+        self.hass.components.mqtt.async_publish(
+            cmd_topic, (keyword + " " + parameters).strip(), qos=0, retain=False
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:526]:
```suggestion
            cmd_topic, f"{keyword} {parameters.strip()}", qos=0, retain=False
```

File: custom_components/openhasp/__init__.py [hunk 2]
```diff
@@ -153,6 +157,15 @@ async def async_setup(hass, config):
     component.async_register_entity_service(
         SERVICE_CLEAR_PAGE, {vol.Optional(ATTR_PAGE): int}, "async_clearpage"
     )
+    component.async_register_entity_service(
+        SERVICE_COMMAND, {
+            vol.Optional(ATTR_COMMAND, default="command"): cv.string,
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:162]:
```suggestion
            vol.Optional(ATTR_COMMAND, default=DEFAULT_COMMAND): cv.string, 
```

File: custom_components/openhasp/__init__.py [hunk 3]
```diff
@@ -504,6 +518,14 @@ async def async_change_page(self, page):
         )
         self.async_write_ha_state()
 
+    async def async_command_wrapper(self, command, keyword, parameters):
+        """Sends commands directly to the plate entity (as a wrapper for MQTT commands sent to hasp/<nodename>/command)"""
+        cmd_topic = f"{self._topic}/" + command
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:523]:
```suggestion
        cmd_topic = f"{self._topic}/{command}"
```

**Review by dgomes — COMMENTED** [review]
When is the command topic not "command" ?

File: custom_components/openhasp/__init__.py (hunk 1 above)
Comment (nagyrobi) [reply custom_components/openhasp/__init__.py:526]: We need to strip the result of `(keyword + " " + parameters)` to ensure that no space remains even if parameters is empty.
`(f"{keyword} {parameters}").strip()` is possible?

**Comment by nagyrobi** [comment]
See in [examples](https://github.com/HASwitchPlate/openHASP-custom-component/issues/50#issuecomment-850961409) _Set Long idle time_. Any other setting applies like this.

File: custom_components/openhasp/__init__.py (hunk 1 above)
Comment (dgomes) [reply custom_components/openhasp/__init__.py:526]: didn't realize you expected leading whitespaces in keyword

Yes that works, even without the ()

**Comment by dgomes** [comment]
"Set long idle time" is an example of something prone to create problems... maybe we should revise that command altogether...

Commands should really be a list of key**words** not json objects...

**Comment by nagyrobi** [comment]
What do you mean? Several commands require json formatted lines as parameters. `jsonl` is the strongest example. "Set long idle time" is just similar to that, but on other (settings) topic.

We need to set idle time 2 times a day, as we don't want the screen to go completely off, just during the night. So during the day the idle2 time is 0, while during the night it's 120. (it's in the [example automations](https://haswitchplate.github.io/openHASP-docs/0.6/integrations/home-assistant/sampl_autom/#turn-on-moodlight-when-backlight-off-and-vice-versa), I'd like to make a blueprint instead, to simplify it)

**Comment by dgomes** [comment]
json objects in **parameters** is absolutely fine, my issue is with **keyword**

**Comment by nagyrobi** [comment]
Let's make the keyword optional?

**Comment by dgomes** [comment]
I say, lets remove the keyword and make the topic always */command

**Comment by nagyrobi** [comment]
But then, how to modify setting? That was the most important point...

**Comment by dgomes** [comment]
lets have a "config" service, but lets discuss that with @fvanroie

**Comment by nagyrobi** [comment]
I can submit a PR of course for a separate service.

I thought that you were more concerned about having too many services as HA admins don't like that on long term. My goal was to have an all-in-one service, which was intended for more advanced use...

**Comment by dgomes** [comment]
If we drop the existing services (change page, etc) in the long run we will not have more then 2 services

It is also important that we create services that can have their parameters sanitised, too much flexibility means we can't check the user input for mistakes...

**Pull request closed without merging**